digitalocean_k8s_centos_7_master.sh

```diff
@@ -50,7 +50,7 @@ PUBLICIP=$(curl -s http://169.254.169.254/metadata/v1/interfaces/public/0/ipv4/a
 echo $PRIVATEIP > /tmp/.ip
 
 # Specify node IP for kubelet.
-echo "Environment=\"KUBELET_EXTRA_ARGS=--node-ip=${PUBLICIP}\"" >> /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+echo "KUBELET_EXTRA_ARGS=--node-ip=${PUBLICIP}" > /etc/default/kubelet
 systemctl daemon-reload
 systemctl restart kubelet
 
```

Comment:
It may be required to update the `kubeadm` config file as well, but I will check this and get back to you.

Usually when using some built-in cloud provider, such as `aws`, [`cloudProvider` key is set to `aws`](https://github.com/kubicorn/bootstrap/blob/master/amazon_k8s_ubuntu_16.04_master.sh#L52). I'm not sure should it be set to `external` and is it even a thing, so I'll check that.

Reply:
I just tried it out, all nodes are stuck in "NotReady", with the master also having these taints:
```
Taints:             node-role.kubernetes.io/master:NoSchedule
                    node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule
```

DigitalOceans 'Cloud-controller' have this error:
`Warning  FailedScheduling  2m (x17 over 3m)  default-scheduler  0/3 nodes are available: 3 node(s) were not ready.`
and these tolerations:
```
Tolerations:     CriticalAddonsOnly
                 node-role.kubernetes.io/master:NoSchedule
                 node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule
                 node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
``` 

My understanding is that the cloud-controller *should* be able to run on the master node, *but* it doesn't.

https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/

Reply:
@Richard87 Are there any additional details when you describe the pods? The taints and toleration look okay to to me.

Have you checked Weave status as mentioned in the previous message? Is it constantly running and do logs look okay?

Reply:
Yeah, I noticed the Weavenet where in CrashLoopBackoff while the master had `--cloud-provider=external` applied, but I deleted the cluster before I looked closer at it ;)

Give me a few minutes! :)

Reply:
Weavenet tolerations:
```
Tolerations:     :NoSchedule
                 node.kubernetes.io/disk-pressure:NoSchedule
                 node.kubernetes.io/memory-pressure:NoSchedule
                 node.kubernetes.io/not-ready:NoExecute
                 node.kubernetes.io/unreachable:NoExecute
```
...
So that explains why Weavenet won't start ;)

Reply:
That might explain why CCM doesn't work, but still unsure how to solve it.

An instance, including Master, will turn Ready once Kubelet reports it's healthy, but without CNI working, it will not be healthy. CCM, even with tolerations, will not be scheduled until node doesn't become ready.

That's totally okay, but why Weave doesn't work is not.

Is there any other errors beside `Failed to get peers`? What's that status of `kube-proxy`?

Reply:
I found this guy: http://tiven.wang/articles/kubernetes-create-cluster-using-kubeadm/

His problem was missing `--pod-network-cidr`  and solved it with `--pod-network-cidr=10.32.0.0/16`

Reply:
That might solve the problem, but we have to try. You need to modify the [kubeadm config file](https://github.com/kubicorn/bootstrap/pull/23/files#diff-01e7451668350e3138ffdce4839fae4cL67) to include the pod network CIDR.

When using kubeadm config files, the key is called little bit different, and is `networking.podSubnet`. It should look like [this one](https://github.com/kubicorn/bootstrap/pull/23/files#diff-01e7451668350e3138ffdce4839fae4cL67).

Reply:
Hi, It's still not working :/

kube-proxy is running nicely...
```
NAME                                                     READY     STATUS             RESTARTS   AGE
coredns-78fcdf6894-bhdvr                                 0/1       Pending            0          5m
coredns-78fcdf6894-rs77p                                 0/1       Pending            0          5m
csi-do-controller-0                                      0/3       Pending            0          5m
digitalocean-cloud-controller-manager-79cff6f759-cx24w   0/1       Pending            0          5m
etcd-test-master-0                                       1/1       Running            0          4m
kube-apiserver-test-master-0                             1/1       Running            0          4m
kube-controller-manager-test-master-0                    1/1       Running            0          4m
kube-proxy-65v5j                                         1/1       Running            0          5m
kube-proxy-tpq45                                         1/1       Running            0          5m
kube-proxy-wzrjn                                         1/1       Running            0          5m
kube-scheduler-test-master-0                             1/1       Running            0          4m
weave-net-ddswz                                          1/2       CrashLoopBackOff   5          5m
weave-net-lkndr                                          1/2       CrashLoopBackOff   5          5m
weave-net-v4xl6                                          1/2       CrashLoopBackOff   5          5m
```
But that's about it that's running :S

Reply:
I changed the CNI to Flannel, but it didn't work either with `--cloud-provider=external` on the master kubelet. 

But removing `--cloud-provider=external` on the master, and both Flannel and Weavenet works.

Reply:
Hmm, strange. Let me check with the DO-CCM team does this sounds okay, and I'll get back to you. :)

Thanks a lot for your hard work on this PR. We really appreciate it! I'll try to update you as soon as possible.